docs/build-transaction-construction.md

```diff
@@ -0,0 +1,210 @@
+    metadataRpc, // must import from client RPC call state_getMetadata
+    nonce: 2,
+    specVersion: 1019,
+    tip: 0,
+    validityPeriod: 240 * 60 // seconds (240 minutes)
+  }
+);
+```
+
+**Construct a signing payload**
+
+```ts
+import { methods, createSigningPayload } from '@substrate/txwrapper';
+
+// See "Construct a transaction offline" for "{...}"
+const unsigned = methods.balances.transferKeepAlive({...}, {...});
+const signingPayload = createSigningPayload(unsigned);
+```
+
+**Serialize a signed transaction**
+
+```ts
+import { createSignedTx } from '@substrate/txwrapper';
+
+// Example code, replace `signWithAlice` with actual remote signer.
+// An example is given here:
+// https://github.com/paritytech/txwrapper/blob/630c38d/examples/index.ts#L50-L68
+const signature = await signWithAlice(signingPayload);
+const signedTx = createSignedTx(unsigned, signature);
+```
+
+**Decode payload types**
+
+You may want to decode payloads to verify their contents prior to submission.
+
+```ts
+import { decode } from '@substrate/txwrapper';
+
+// Decode an unsigned tx
+const txInfo = decode(unsigned, { metadata });
+
+// Decode a signing payload
+const txInfo = decode(signingPayload, { metadata });
+
+// Decode a signed tx
+const txInfo = decode(signedTx, { metadata });
+```
+
+**Check a transaction's hash**
+
+```ts
+import { getTxHash } from ‘@substrate/txwrapper’;
+const txHash = getTxHash(signedTx);
+```
+
+## Submitting a Signed Payload
+
+There are several ways to submit a signed payload:
+
+1. Signer CLI (`yarn run:signer submit --tx <signed-transaction> --ws <endpoint>`)
+1. [Substrate API Sidecar](build-node-interaction#substrate-api-sidecar)
+1. [RPC](build-node-interaction#polkadot-rpc) with `author_submitExtrinsic` or
+`author_submitAndWatchExtrinsic`, the latter of which will subscribe you to events to be notified
+as a transaction gets validated and included in the chain.
+
+## Notes
+
+Some addresses to use in the examples. See
+[Subkey documentation](https://substrate.dev/docs/en/next/development/tools/subkey).
+
+```bash
+$ subkey --network polkadot generate
+Secret phrase `pulp gaze fuel ... mercy inherit equal` is account:
+  Secret seed:      0x57450b3e09ba4598 ... ... ... ... ... ... ... .. 219756eeba80bb16
+  Public key (hex): 0x2ca17d26ca376087dc30ed52deb74bf0f64aca96fe78b05ec3e720a72adb1235
+  Account ID:       0x2ca17d26ca376087dc30ed52deb74bf0f64aca96fe78b05ec3e720a72adb1235
+  SS58 Address:     121X5bEgTZcGQx5NZjwuTjqqKoiG8B2wEAvrUFjuw24ZGZf2
+
+$ subkey --network polkadot generate
+Secret phrase `exercise auction soft ... obey control easily` is account:
+  Secret seed:      0x5f4bbb9fbb69261a ... ... ... ... ... ... ... .. 4691ed7d1130fbbd
+  Public key (hex): 0xda04de6cd781c98acf0693dfb97c11011938ad22fcc476ed0089ac5aec3fe243
+  Account ID:       0xda04de6cd781c98acf0693dfb97c11011938ad22fcc476ed0089ac5aec3fe243
+  SS58 Address:     15vrtLsCQFG3qRYUcaEeeEih4JwepocNJHkpsrqojqnZPc2y
+```
```

Comment:
If these are examples, why are the secret seeds and phrases abbreviated?  Just to keep the line length reasonable?

Reply:
Kirill brought up a good point that there have been some "attacks" (more like careless mistakes) of people using private keys from documentation. So private keys just shouldn't be available for the taking.

Reply:
Ahh interesting!  Makes sense.